src/V3Width.cpp

```diff
@@ -1528,7 +1528,8 @@ class WidthVisitor final : public VNVisitor {
         case VAttrType::DIM_SIZE: {
             UASSERT_OBJ(nodep->fromp() && nodep->fromp()->dtypep(), nodep, "Unsized expression");
             AstNodeDType* const dtypep = nodep->fromp()->dtypep();
-            if (VN_IS(dtypep, QueueDType)) {
+            if (VN_IS(dtypep, QueueDType)
+                    || VN_IS(dtypep, DynArrayDType)) {
```

Comment:
This makes `$size`, `$right`, `$high` work for dynarray, but with the same limitation as for queue: the second argument completely ignored and only first dimension is considered.